### PR TITLE
Make suspend template callable, add suspend examples

### PR DESCRIPTION
**Pull request by elliotgunton**
* Fixes ability to add raw model Template in workflow.py

Related to #594 - should we add public API for `workflow.add_template`, otherwise I have to use `workflow._add_sub`

### CODE REVIEW

**Comment by flaviuvadan** [comment]
> * Fixes ability to add raw model Template in workflow.py
> 
> Related to #594 - should we add public API for `workflow.add_template`, otherwise I have to use `workflow._add_sub`

Yea, I think `add_template` sounds reasonable!

File: examples/workflows/upstream/suspend-template.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from hera.workflows import Container, Step, Steps, Suspend, Workflow
+
+with Workflow(
+    generate_name="suspend-template-",
+    entrypoint="suspend",
+) as w:
+    whalesay = Container(name="whalesay", image="docker/whalesay", command=["cowsay"], args=["hello world"])
+
+    approve = Suspend(name="approve")
```
[the remark this answers is not in the document]
Comment (elliotgunton) [reply examples/workflows/upstream/suspend-template.py:9]: Yeah I was surprised I couldn't call it in the context actually. Seems reasonable to me to make it a callable template mixin

File: examples/workflows/upstream/suspend-template-outputs.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+) as w:
+    whalesay = Container(
+        name="whalesay",
+        image="docker/whalesay",
+        command=["cowsay"],
+        inputs=[Parameter(name="message")],
+        args=["{{inputs.parameters.message}}"],
+    )
+
+    w._add_sub(
```
Comment (samj1912) [inline examples/workflows/upstream/suspend-template-outputs.py:16]: We should not advertise private commands in public examples.

**Comment by elliotgunton** [comment]
> Yea, I think add_template sounds reasonable!

File: examples/workflows/upstream/suspend-template.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    delay = Suspend(name="delay", duration=20)
+
+    with Steps(name="suspend"):
+        whalesay(name="build")
+        approve()
+        delay()
+        whalesay(name="release")
```
Comment (flaviuvadan) [inline examples/workflows/upstream/suspend-template.py:16]: this looks amazing :)

**Review by flaviuvadan — APPROVED** [review]

**Pull request merged**